internal/gitlab/codequality.go

```diff
@@ -0,0 +1,117 @@
+// Package gitlab generates GitLab Code Quality reports.
+package gitlab
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/yamlfmt"
+)
+
+// CodeQuality represents a single code quality finding.
+//
+// Documentation: https://docs.gitlab.com/ee/ci/testing/code_quality.html#code-quality-report-format
+type CodeQuality struct {
+	Description string
+	Name        string
+	Fingerprint string
+	Severity    Severity
+	Path        string
+}
+
+// NewCodeQuality creates a new CodeQuality object from a yamlfmt.FileDiff.
+//
+// If the file did not change, i.e. the diff is empty, an empty struct and false is returned.
+func NewCodeQuality(diff yamlfmt.FileDiff) (CodeQuality, bool) {
+	if !diff.Diff.Changed() {
+		return CodeQuality{}, false
+	}
+
+	return CodeQuality{
+		Description: "Not formatted correctly, run yamlfmt to resolve.",
+		Name:        "yamlfmt",
+		Fingerprint: fingerprint(diff),
+		Severity:    Major,
+		Path:        diff.Path,
+	}, true
+}
+
+// Marshals a CodeQuality object into JSON.
```

Comment:
For consistency with the rest of the file:
```suggestion
// MarshalJSON marshals a CodeQuality object into JSON.
```
(I don't actually follow this rule myself throughout the codebase like I should)

Reply:
I usually do follow this rule religiously. I guess I didn't properly check the documentation produced by my AI ~overlord~ apprentice. 🤷